src/main/java/org/broadinstitute/consent/http/service/NihServiceAPI.java

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.consent.http.service;
+
+import org.broadinstitute.consent.http.enumeration.ResearcherFields;
+import org.broadinstitute.consent.http.models.NIHUserAccount;
+import org.broadinstitute.consent.http.models.ResearcherProperty;
+import org.broadinstitute.consent.http.service.users.handler.ResearcherAPI;
+
+import javax.ws.rs.BadRequestException;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+
+public class NihServiceAPI implements NihAuthApi {
+
+    private ResearcherAPI researcherAPI;
+
+    public NihServiceAPI(ResearcherAPI researcherApi) {
+        this.researcherAPI = researcherApi;
+    }
+
+    @Override
+    public List<ResearcherProperty> authenticateNih(NIHUserAccount nihAccount, Integer userId) throws BadRequestException{
+        if (!nihAccount.getNihUsername().isEmpty()) {
```

Comment:
could NihUsername be null? if this happen isEmpty will throw NPE

Reply:
You're right, added an additional condition to save that situation .